app/models/api_caller.rb

```diff
@@ -0,0 +1,52 @@
+require 'open-uri'
+require 'JSON'
+
+class ApiCaller
+  def initialize
+  end
+
+  def request(nonos_array, search_term)
+    query = generate_query(nonos_array, search_term)
+    s = "http://api.yummly.com/v1/api/recipes?_app_id=#{ENV["APPLICATION_ID"]}&_app_key=#{ENV["APPLICATION_KEY"]}#{query}"
+    response = open(s)
+    doc = ""
+    response.each do |line|
+      doc << line
+    end
+    response.close
+
+    JSON.parse(doc)
+  end
+
+  def generate_query(nonos_array, search_term)
+    encoded_search_term = search_term.gsub(/\s+/, "+").downcase
```

Comment:
You may need to do something slightly more sophisticated here. For example a string like "crème brûlée" might cause trouble other than its space. You can use URI.encode_www_form_component - see http://ruby-doc.org/stdlib-1.9.3/libdoc/uri/rdoc/URI.html#method-c-encode_www_form_component